app.py

```diff
@@ -124,20 +125,92 @@ def page2():
         hv.opts.Points(tools=["hover"], alpha=0.1, hover_alpha=0.2, size=10)
     )
 
-    hist_panel = pn.bind(make_hist_panel, log_y=log_y_checkbox, plot_width=plot_width)
+    hist_panel = pn.bind(make_hist_panel, log_y=log_y_checkbox)
 
     plot_options = pn.Column(
         pn.pane.Markdown("## Plot Options"),
         log_y_checkbox,
-        threshold,
     )
 
     return pn.Column(main << time_hist << site_hist, hist_panel, plot_options)
 
 
+def page3():
+    df_edges["parent_time_right"] = df_edges["parent_time"]
+    lines = hv.Segments(
+        df_edges, kdims=["left", "parent_time", "right", "parent_time_right"]
+    )
+    range_stream = hv.streams.RangeXY(source=lines)
+    streams = [range_stream]
+    filtered = lines.apply(filter_points, streams=streams)
+    hover = filtered.apply(hover_points, threshold=threshold)
+    shaded = hd.datashade(filtered, streams=streams)
+    hover_tool = bkm.HoverTool(
+        tooltips=[
+            ("child", "@child"),
+            ("parent", "@parent"),
+            ("span", "@span"),
+            ("branch_length", "@branch_length"),
+        ]
+    )
+    main = (shaded * hover).opts(
+        hv.opts.Segments(
+            tools=[hover_tool],
+            width=plot_width,
+            height=plot_height,
+            xlabel="Position",
+            ylabel="Time",
+        )
+    )
+
+    return pn.Column(main)
+
+
+def page4():
+    node_id_input = pn.widgets.TextInput(value="", name="Node ID")
+    df_edges["parent_time_right"] = df_edges["parent_time"]
+    tabulator = pn.widgets.Tabulator(show_index=False)
+
+    def plot_data(node_id):
+        if node_id:
```

Comment:
Minor nit - I'm not a fan of "falsey" evaluation, although it's something of a religious question.

I think it's better to be explicit here - I think this should be ``if len(node_id) > 0``